c7n/resources/efs.py

```diff
@@ -137,3 +138,48 @@ def unmount_filesystems(self, resources):
             for t in client.describe_mount_targets(
                     FileSystemId=r['FileSystemId'])['MountTargets']:
                 client.delete_mount_target(MountTargetId=t['MountTargetId'])
+
+
+@ElasticFileSystem.action_registry.register('configure-lifecycle-policy')
+class ConfigureLifecycle(BaseAction):
+    """Enable/disable lifecycle policy for efs.
+
+    :example:
+
+      .. code-block:: yaml
+
+      - name: efs-apply-lifecycle
+        resource: efs
+        actions:
+          - type: configure-lifecycle-policy
+            state: enable
+            rule:
+                - 'TransitionToIA': 'AFTER_7_DAYS'
+
+    """
+    schema = type_schema(
+        'configure-lifecycle-policy',
+        state={'enum': ['enable', 'disable']},
+        rule={'type': 'array'},
+        required=['state'])
+
+    permissions = ('elasticfilesystem:PutLifecycleConfiguration',)
+
+    def validate(self):
+        if self.data.get('state') == 'enable' and 'rule' not in self.data:
```

Comment:
if we're doing a jsonschema pass through on lifecycle polices, we should validate them with shape_validate (see aws.py module) against the sdk metadata.

Reply:
Thanks, I have added code to validate lifecycle policy rules as per your suggestion. I have kept the other two validations for the following reasons:
1) `state: enable` should not be used without any rule specified
2) `state: disable` assumes the rule to be an empty list
Let me know if this still requires changes! Thanks